lib/galaxy/managers/collections.py

```diff
@@ -802,7 +802,7 @@ def __get_history_collection_instance(
             raise RequestParameterInvalidException("History dataset collection association not found")
         # TODO: that sure looks like a bug, we can't check ownership using the history of the object we're checking ownership for ...
         history = getattr(trans, "history", collection_instance.history)
-        if check_ownership:
+        if check_ownership and collection_instance.history.published is False:
```

Comment:
That is what `check_accessible=True` is supposed to do, this is the check that is applied if we're doing stuff for which you need to be owner. I think you need to find the API routes that are problematic, and then check if we're modifying the collection, which means `check_ownership="True"`, or if we're just viewing then `check_accessible=True` is fine

Reply:
Ah i see, that makes a lot of sense, I should instead look at whether any modifications are being made or if we're just viewing, and check ownership/accessibility accordingly..

Reply:
Updated the changes, does this make more sense?: I removed the `check_ownership=True` flag since this api (`/api/dataset_collections/{hdca_id}/contents/{parent_id}`) doesn't modify a collection. This also works as it allows you to access collections from public histories and doesn't if history is private.